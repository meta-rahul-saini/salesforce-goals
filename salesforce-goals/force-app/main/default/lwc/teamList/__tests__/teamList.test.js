import { createElement } from 'lwc';
import TeamList from 'c/teamList';
import { getListUi } from 'lightning/uiListApi';
import { registerTestWireAdapter, registerLdsTestWireAdapter, registerApexTestWireAdapter } from '@salesforce/sfdx-lwc-jest';
import getMembersList from '@salesforce/apex/MemberController.getMembersList';
import {
    MessageContext
} from 'lightning/messageService';

const flushPromises = () =>
    new Promise((resolve) => setImmediate(resolve));

// Mock realistic data
const mockGetTeamList = require('./data/getTeamList.json');
const mockGetMemberList = require('./data/getMemberList.json');
  
// Register a standard test wire adapter.
const getTeamListUiAdapter = registerLdsTestWireAdapter(getListUi);
const getMemberListApexAdapter = registerApexTestWireAdapter(getMembersList);

registerTestWireAdapter(MessageContext);

const getElement =  (props={})=>{
    const element = createElement('c-team-list', {
    is: TeamList,
    });
    Object.assign(element, props)
    document.body.appendChild(element);
    return element;
}

describe('c-team-list', async () => {
    afterEach(() => {
        // The jsdom instance is shared across test cases in a single file so reset the DOM
        while (document.body.firstChild) {
            document.body.removeChild(document.body.firstChild);
        }
    });

    it('should render', async () => {
        const element = getElement();        
        expect(element).toMatchSnapshot();
    });

    it('should render all the teams in options in combobox', async () => {
        const element = getElement();
        getTeamListUiAdapter.emit(mockGetTeamList);
        getMemberListApexAdapter.emit(mockGetMemberList);
        await flushPromises();        
        const lightningEL =  element.shadowRoot.querySelector('lightning-combobox');
        console.log('lightningEL: ', lightningEL.options.length == 2);
        expect(lightningEL.length === mockGetTeamList.records.records.length);
        expect(element).toMatchSnapshot();
    });

    it('on select an options the component should render all the available members in that team', async () => {
        const element = getElement();
        getTeamListUiAdapter.emit(mockGetTeamList);
        getMemberListApexAdapter.emit(mockGetMemberList);
        await flushPromises();
        const lightningEL =  element.shadowRoot.querySelector('lightning-combobox');
        lightningEL.value = mockGetTeamList.records.records[0].fields.Name.value;
        await flushPromises();
        expect(element).toMatchSnapshot();
    });
});