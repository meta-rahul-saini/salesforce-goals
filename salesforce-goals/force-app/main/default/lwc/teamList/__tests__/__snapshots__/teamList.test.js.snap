// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-team-list on select an options 1`] = `
<c-team-list>
  #shadow-root(open)
    <lightning-card>
      #shadow-root(open)
        <slot />
        <slot
          name="actions"
        />
        <slot
          name="footer"
        />
        <slot
          name="title"
        />
      <h3
        slot="title"
      >
        <lightning-icon>
          #shadow-root(open)
        </lightning-icon>
        Member List
      </h3>
      <div
        class="slds-card-body"
      >
        <lightning-combobox>
          #shadow-root(open)
        </lightning-combobox>
      </div>
      <div>
        <hr />
        <lightning-card>
          #shadow-root(open)
            <slot />
            <slot
              name="actions"
            />
            <slot
              name="footer"
            />
            <slot
              name="title"
            />
          <dl
            class="slds-list_horizontal slds-wrap"
          >
            <dt
              class="slds-item_label slds-text-color_weak slds-truncate"
              title="Name"
            >
              Name:
            </dt>
            <dd
              class="slds-item_detail slds-truncate"
              title="Description for first label"
            >
              Raghu
            </dd>
            <dt
              class="slds-item_label slds-text-color_weak slds-truncate"
              title="Team"
            >
              Team
            </dt>
            <dd
              class="slds-item_detail slds-truncate"
              title="Description for second label"
            >
              a0B2w000003hMtkEAE
            </dd>
            <dt
              class="slds-item_label slds-text-color_weak slds-truncate"
              title="Skills"
            >
              Skills
            </dt>
            <dd
              class="slds-item_detail slds-truncate"
              title="Description for second label"
            >
              AI;ML
            </dd>
          </dl>
        </lightning-card>
      </div>
      <div>
        <hr />
        <lightning-card>
          #shadow-root(open)
            <slot />
            <slot
              name="actions"
            />
            <slot
              name="footer"
            />
            <slot
              name="title"
            />
          <dl
            class="slds-list_horizontal slds-wrap"
          >
            <dt
              class="slds-item_label slds-text-color_weak slds-truncate"
              title="Name"
            >
              Name:
            </dt>
            <dd
              class="slds-item_detail slds-truncate"
              title="Description for first label"
            >
              Rahul
            </dd>
            <dt
              class="slds-item_label slds-text-color_weak slds-truncate"
              title="Team"
            >
              Team
            </dt>
            <dd
              class="slds-item_detail slds-truncate"
              title="Description for second label"
            >
              a0B2w000003hMtkEAE
            </dd>
            <dt
              class="slds-item_label slds-text-color_weak slds-truncate"
              title="Skills"
            >
              Skills
            </dt>
            <dd
              class="slds-item_detail slds-truncate"
              title="Description for second label"
            >
              ML;DS
            </dd>
          </dl>
        </lightning-card>
      </div>
      <div>
        <hr />
        <lightning-card>
          #shadow-root(open)
            <slot />
            <slot
              name="actions"
            />
            <slot
              name="footer"
            />
            <slot
              name="title"
            />
          <dl
            class="slds-list_horizontal slds-wrap"
          >
            <dt
              class="slds-item_label slds-text-color_weak slds-truncate"
              title="Name"
            >
              Name:
            </dt>
            <dd
              class="slds-item_detail slds-truncate"
              title="Description for first label"
            >
              Rakesh
            </dd>
            <dt
              class="slds-item_label slds-text-color_weak slds-truncate"
              title="Team"
            >
              Team
            </dt>
            <dd
              class="slds-item_detail slds-truncate"
              title="Description for second label"
            >
              a0B2w000003hMtkEAE
            </dd>
            <dt
              class="slds-item_label slds-text-color_weak slds-truncate"
              title="Skills"
            >
              Skills
            </dt>
            <dd
              class="slds-item_detail slds-truncate"
              title="Description for second label"
            >
              AI
            </dd>
          </dl>
        </lightning-card>
      </div>
    </lightning-card>
</c-team-list>
`;

exports[`c-team-list should render 1`] = `
<c-team-list>
  #shadow-root(open)
    <lightning-card>
      #shadow-root(open)
        <slot />
        <slot
          name="actions"
        />
        <slot
          name="footer"
        />
        <slot
          name="title"
        />
      <h3
        slot="title"
      >
        <lightning-icon>
          #shadow-root(open)
        </lightning-icon>
        Member List
      </h3>
      <div
        class="slds-card-body"
      >
        <lightning-combobox>
          #shadow-root(open)
        </lightning-combobox>
      </div>
    </lightning-card>
</c-team-list>
`;

exports[`c-team-list should render all the teams in options in combobox 1`] = `
<c-team-list>
  #shadow-root(open)
    <lightning-card>
      #shadow-root(open)
        <slot />
        <slot
          name="actions"
        />
        <slot
          name="footer"
        />
        <slot
          name="title"
        />
      <h3
        slot="title"
      >
        <lightning-icon>
          #shadow-root(open)
        </lightning-icon>
        Member List
      </h3>
      <div
        class="slds-card-body"
      >
        <lightning-combobox>
          #shadow-root(open)
        </lightning-combobox>
      </div>
      <div>
        <hr />
        <lightning-card>
          #shadow-root(open)
            <slot />
            <slot
              name="actions"
            />
            <slot
              name="footer"
            />
            <slot
              name="title"
            />
          <dl
            class="slds-list_horizontal slds-wrap"
          >
            <dt
              class="slds-item_label slds-text-color_weak slds-truncate"
              title="Name"
            >
              Name:
            </dt>
            <dd
              class="slds-item_detail slds-truncate"
              title="Description for first label"
            >
              Raghu
            </dd>
            <dt
              class="slds-item_label slds-text-color_weak slds-truncate"
              title="Team"
            >
              Team
            </dt>
            <dd
              class="slds-item_detail slds-truncate"
              title="Description for second label"
            >
              a0B2w000003hMtkEAE
            </dd>
            <dt
              class="slds-item_label slds-text-color_weak slds-truncate"
              title="Skills"
            >
              Skills
            </dt>
            <dd
              class="slds-item_detail slds-truncate"
              title="Description for second label"
            >
              AI;ML
            </dd>
          </dl>
        </lightning-card>
      </div>
      <div>
        <hr />
        <lightning-card>
          #shadow-root(open)
            <slot />
            <slot
              name="actions"
            />
            <slot
              name="footer"
            />
            <slot
              name="title"
            />
          <dl
            class="slds-list_horizontal slds-wrap"
          >
            <dt
              class="slds-item_label slds-text-color_weak slds-truncate"
              title="Name"
            >
              Name:
            </dt>
            <dd
              class="slds-item_detail slds-truncate"
              title="Description for first label"
            >
              Rahul
            </dd>
            <dt
              class="slds-item_label slds-text-color_weak slds-truncate"
              title="Team"
            >
              Team
            </dt>
            <dd
              class="slds-item_detail slds-truncate"
              title="Description for second label"
            >
              a0B2w000003hMtkEAE
            </dd>
            <dt
              class="slds-item_label slds-text-color_weak slds-truncate"
              title="Skills"
            >
              Skills
            </dt>
            <dd
              class="slds-item_detail slds-truncate"
              title="Description for second label"
            >
              ML;DS
            </dd>
          </dl>
        </lightning-card>
      </div>
      <div>
        <hr />
        <lightning-card>
          #shadow-root(open)
            <slot />
            <slot
              name="actions"
            />
            <slot
              name="footer"
            />
            <slot
              name="title"
            />
          <dl
            class="slds-list_horizontal slds-wrap"
          >
            <dt
              class="slds-item_label slds-text-color_weak slds-truncate"
              title="Name"
            >
              Name:
            </dt>
            <dd
              class="slds-item_detail slds-truncate"
              title="Description for first label"
            >
              Rakesh
            </dd>
            <dt
              class="slds-item_label slds-text-color_weak slds-truncate"
              title="Team"
            >
              Team
            </dt>
            <dd
              class="slds-item_detail slds-truncate"
              title="Description for second label"
            >
              a0B2w000003hMtkEAE
            </dd>
            <dt
              class="slds-item_label slds-text-color_weak slds-truncate"
              title="Skills"
            >
              Skills
            </dt>
            <dd
              class="slds-item_detail slds-truncate"
              title="Description for second label"
            >
              AI
            </dd>
          </dl>
        </lightning-card>
      </div>
    </lightning-card>
</c-team-list>
`;
